sirepo/job_supervisor.py

```diff
@@ -710,11 +708,19 @@ def _set_error(op, compute_job_serial):
                 raise
             # There was a timeout getting the run started. Set the
             # error and let the user know. The timeout has destroyed
-            # the op so don't need to in _set_error
-            _set_error(None, c)
+            # the op so don't need to
+            _set_error(c, o.internal_error)
             return self._status_reply(req)
-        except Exception:
-            _set_error(o, c)
+        except Exception as e:
+            # _run destroys in the happy path (never got to _run here)
+            o.destroy(cancel=False)
+            if isinstance(e, sirepo.util.SRException) and\
+               e.sr_args.params.get('nonJobExc'):
+                self.__db_update(
```

Comment:
One line is fine in this case: self.__db_update(status=s).

Is the update necessary?

Reply:
Yes here is the scenario:
1) status='missing'
2) user clicks "start new simulation"
3) A new db will be initialized with status='pending'  ([line 680](https://github.com/radiasoft/sirepo/blob/1360b687a319208f17394c88c074bc4033717f5a/sirepo/job_supervisor.py#L680)
4. User must login so an SRException is raised
5. This code is executed setting the status to the old status (missing). Which is the status of the job in the run_dir.
5. The login is succesful and the GUI does a runStatus. The status comes back 'missing' which is correct.

Thinking about this more. Maybe we want to restore the entire db record not just the status? Other fields like computeJobSerial are now incorrect.

Reply:
> Maybe we want to restore the entire db record not just the status? 

Yes, this is an interesting problem. The login is happening in connection with a Job, because that's the object we have for connecting to the request. Reality is that we just want to login, and the request has nothing to do with a job. 

Do you think we could handle this request separately so that we don't update the job state at all?

Reply:
https://github.com/radiasoft/sirepo/pull/2523/commits/362eac890cb9e8509e8f41fa1726438eb673fb90